File: easy/3471_find_the_largest_almost_missing_integer/solution.cpp
#include <unordered_map>
#include <vector>

class Solution {
public:
  int largestInteger(std::vector<int> &nums, int k) {
    if (k == static_cast<int>(nums.size()))
      return *max_element(nums.begin(), nums.end());

    std::unordered_map<int, int> numsCounter;
    for (const int &num : nums)
      ++numsCounter[num];

    if (k == 1) {
      int maxNum = -1;

      for (const auto &[num, count] : numsCounter) {
        if (count == 1)
          maxNum = std::max(maxNum, num);
      }

      return maxNum;
    }

    if (numsCounter[nums[0]] == 1 && numsCounter[nums.back()] == 1)
      return (nums[0] > nums.back()) ? nums[0] : nums.back();
    else if (numsCounter[nums[0]] == 1)
      return nums[0];
    else if (numsCounter[nums.back()] == 1)
      return nums.back();

    return -1;
  }
};

File: easy/3471_find_the_largest_almost_missing_integer/solution.py
from collections import Counter


class Solution:
    def largestInteger(self, nums: list[int], k: int) -> int:
        if k == len(nums):
            return max(nums)

        numsCounter = Counter(nums)

        return max(
            (
                num
                for num in (
                    numsCounter.keys() if k == 1 else (nums[0], nums[-1])
                )
                if numsCounter[num] == 1
            ),
            default=-1,
        )
